src/Data/Patch/MapWithMove.hs

```diff
@@ -2,178 +2,199 @@
 {-# LANGUAGE DeriveTraversable #-}
 {-# LANGUAGE FlexibleContexts #-}
 {-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
 {-# LANGUAGE LambdaCase #-}
 {-# LANGUAGE MultiParamTypeClasses #-}
 {-# LANGUAGE PatternGuards #-}
+{-# LANGUAGE PatternSynonyms #-}
 {-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE StandaloneDeriving #-}
 {-# LANGUAGE TemplateHaskell #-}
 {-# LANGUAGE TypeApplications #-}
 {-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE ViewPatterns #-}
 
 -- | 'Patch'es on 'Map' that can insert, delete, and move values from one key to
 -- another
-module Data.Patch.MapWithMove where
+module Data.Patch.MapWithMove
+  ( PatchMapWithMove
+    ( PatchMapWithMove
+    , unPatchMapWithMove
+    , ..
+    )
+  , patchMapWithMove
+  , patchMapWithMoveInsertAll
+  , insertMapKey
+  , moveMapKey
+  , swapMapKey
+  , deleteMapKey
+  , unsafePatchMapWithMove
+  , patchMapWithMoveNewElements
+  , patchMapWithMoveNewElementsMap
+  , patchThatSortsMapWith
+  , patchThatChangesAndSortsMapWith
+  , patchThatChangesMap
 
+  -- * Node Info
+  , NodeInfo
+    ( NodeInfo
+    , _nodeInfo_to
+    , _nodeInfo_from
+    , ..
+    )
+  , bitraverseNodeInfo
+  , nodeInfoMapFrom
+  , nodeInfoMapMFrom
+  , nodeInfoSetTo
+
+  -- * From
+  , From
+    ( From_Insert
+    , From_Delete
+    , From_Move
+    , ..
+    )
+  , bitraverseFrom
+
+  -- * To
+  , To
+  ) where
+
+import Data.Coerce
+import Data.Kind (Type)
 import Data.Patch.Class
+import Data.Patch.MapWithPatchingMove (PatchMapWithPatchingMove(..), To)
+import qualified Data.Patch.MapWithPatchingMove as PM -- already a transparent synonym
 
-import Control.Arrow
-import Control.Lens hiding (from, to)
-import Control.Monad.Trans.State
-import Data.Foldable
-import Data.Function
+import Control.Lens
 import Data.List
 import Data.Map (Map)
 import qualified Data.Map as Map
-import Data.Maybe
+import Data.Proxy
 #if !MIN_VERSION_base(4,11,0)
 import Data.Semigroup (Semigroup (..))
 #endif
-import qualified Data.Set as Set
-import Data.These (These(..))
-import Data.Tuple
+import Data.Traversable (foldMapDefault)
 
 -- | Patch a Map with additions, deletions, and moves.  Invariant: If key @k1@
 -- is coming from @From_Move k2@, then key @k2@ should be going to @Just k1@,
 -- and vice versa.  There should never be any unpaired From/To keys.
-newtype PatchMapWithMove k v = PatchMapWithMove
-  { -- | Extract the internal representation of the 'PatchMapWithMove'
-    unPatchMapWithMove :: Map k (NodeInfo k v)
+newtype PatchMapWithMove k (v :: Type) = PatchMapWithMove'
```

Comment:
PolyKinds was mischievously generalizing this over things for which `PatchTarget (Proxy v)` is undefined. Oh type families...